src/Search/index.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+export interface SearchProps {
+  placeholder?: string
+  color?: string
+  background?: string
+  debounce?: number
+  onSubmit: (value: string | null) => void
+}
+
+interface SearchWrapperProps extends React.DOMAttributes<HTMLFormElement> {
+  color: string
+}
+
+interface SearchInputProps extends React.DOMAttributes<HTMLInputElement> {
+  color: string
+}
+
+interface SearchIconProps extends React.DOMAttributes<HTMLButtonElement> {
+  color: string
+}
+
+const Search: React.FC<SearchProps> = ({
+  placeholder,
+  color = 'black',
+  background = 'transparent',
+  debounce = 250,
+  onSubmit,
+}) => {
+  const [search, setSearch] = useState<string | null>(null)
+  const [hasSubmitted, setHasSubmitted] = useState(true)
+  const buttonRef = useRef<HTMLButtonElement | null>(null)
+
+  const handleSearchChange: React.ChangeEventHandler<HTMLInputElement> =
+    useCallback(
```

Comment:
never used this method. interesting

Reply:
if you define a function in a component you should probably be using `useCallback`. It prevents garbage collection spam by reducing the need to constantly recreate a function on new renders